src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -55,19 +55,19 @@ public DateHeaderValueManager()
         /// Returns a value representing the current server date/time for use in the HTTP "Date" response header
         /// in accordance with http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18
         /// </summary>
-        /// <returns>The value.</returns>
-        public virtual string GetDateHeaderValue()
+        /// <returns>The value in string and byte[] format.</returns>
+        public DateHeaderValues GetDateHeaderValues()
         {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _dateValue;
-        }
+            if (!_hadRequestsSinceLastTimerTick)
+            {
+                PrepareDateValues();
+            }
 
-        public byte[] GetDateHeaderValueBytes()
-        {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _activeDateBytes ? _dateBytes0 : _dateBytes1;
+            return new DateHeaderValues()
+            {
+                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
```

Comment:
Not specific to this change, but I'm unclear how the current usage of _dateBytes0 and _dateBytes1 is safe.  It appears that the code is trying to use one of the arrays as the current known-good and the other array as a scratch, and they then get switched when the scratch is updated.  But what's to prevent one thread from getting handed one of the arrays, that thread getting swapped out, lots of stuff happening, and then that thread getting resumed as another thread is encoding a new value into that same array?  Do we know that the encoding process is going to leave the data in the array in a non-corrupted state from the perspective of having valid data?